spec/controllers/application_controller_spec.rb

```diff
@@ -16,6 +16,25 @@ def index
     end
   end
 
+  describe '#cache_issuer_in_cookie' do
+    controller do
+      def index
+        render plain: 'Hello'
+      end
+    end
+
+    let(:sp) { create(:service_provider, issuer: 'urn:gov:gsa:openidconnect:sp:test_cookie') }
+    before do
+      allow_any_instance_of(ApplicationController).to receive(:current_sp).and_return(sp)
+    end
+
+    it 'sets headers to disable cache' do
+      get :index
```

Comment:
What if we did a spec that passed in a `request_id` param and then we could just create an ServiceProviderRequest with a corresponding issuer, instead of having to stub?

Reply:
That would be appropriate with the oidc AuthorizationController, but this is just looking at the fact that any action triggers the setting of the action. Should I add that test in as well?

Reply:
It was just a suggestion of how to simplify the test by removing stubbing --- I don't feel that strongly about it as long as the core logic is tested. But it would be good to make sure that from a feature test for either OIDC or SAML that the cookie gets set, now that you mention it (I don't feel that strongly about it)

Reply:
I've added a simple spec on both the OIDC and SAML feature specs.